src/content-helper/editor-sidebar/smart-linking/component.tsx

```diff
@@ -445,78 +445,80 @@ export const SmartLinkingPanel = ( {
 				return;
 			}
 
-			if ( block.originalContent ) {
-				const blockContent: string = block.originalContent;
-				const doc = new DOMParser().parseFromString( blockContent, 'text/html' );
+			if ( ! block.originalContent ) {
+				return;
+			}
```

Comment:
There might be some cases where the block doesn't have `originalContent` prop (for example a Group block). The getBlockContent might still return a value, since it always returns a string, hence this validation being required. 

@coderabbitai resolve